core/src/services/dropbox/core.rs

```diff
@@ -111,6 +111,23 @@ impl DropboxCore {
         self.client.send(request).await
     }
 
+    pub async fn dropbox_get_stat(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let url = "https://api.dropboxapi.com/2/files/get_metadata".to_string();
+        let args = DropboxMetadataArgs {
+            path: build_rooted_abs_path(&self.root, path),
+            ..Default::default()
+        };
+
+        let bs = Bytes::from(serde_json::to_string(&args).map_err(new_json_serialize_error)?);
+
+        let request = self
+            .build_auth_header(Request::post(&url))
```

Comment:
Please build request first and calling `self.core.sign()` instead.

Reply:
I chose the name `build_auth_header` because I'm not sure if the `sign` is a good name for this action.

In dropbox, what are all we need is to add to token to the header. the token is not about the body content, we also don't need a make a signature action to build the header. So if it is semantic for Dropbox service

Reply:
> In dropbox, what are all we need is to add to token to the header.

This is the `sign`.

Reply:
This is not a blocker, we can do this in another PR.